app/scripts/ui.js

```diff
@@ -95,10 +94,14 @@ async function queryCurrentActiveTab (windowType) {
     extension.tabs.query({ active: true, currentWindow: true }, (tabs) => {
       const [activeTab] = tabs
       const { title, url } = activeTab
-      const { hostname: origin, protocol } = url ? urlUtil.parse(url) : {}
-      resolve({
-        title, origin, protocol, url,
-      })
+
+      let urlData = {}
+      try {
+        urlData = new URL(url)
+      } catch (_) {}
+      const { origin, protocol } = urlData
```

Comment:
It would be nice to avoid using exceptions for control flow - we could do something like this instead:

```suggestion
      const { origin, protocol } = url ? new URL(url) : {}
```

That way it continues to blow up if `url` is truthy but still throws.

Reply:
Ah, the difference here is that the `URL` constructor throws exceptions far more liberally than Node's `url.parse`. I would've kept the latter, but its result doesn't have an `origin` property 😑

As far as I can tell, `url.parse` only throws for non-string args and invalid `auth` properties: https://nodejs.org/api/url.html#url_url_parse_urlstring_parsequerystring_slashesdenotehost

Whereas:
```javascript
> new URL('foo')
Thrown:
{ TypeError [ERR_INVALID_URL]: Invalid URL: foo
    at onParseError (internal/url.js:241:17)
    at new URL (internal/url.js:319:5) input: 'foo' }
```

Reply:
Basically, I don't want to cause the UI to blow up on boot in cases where it currently doesn't.

Reply:
Fair enough, though in this case I don't think it's possible `url` to be an invalid URL, or at least not without a Chrome bug. This is given to us directly by the Chrome API.

Reply:
Hm, yeah, fair enough, I went with your suggestion!